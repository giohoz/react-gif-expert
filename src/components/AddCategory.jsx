import { useState } from "react";
import  PropTypes  from "prop-types";

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) =>{
        //console.log(target.value);
        setInputValue(target.value);
    }

    const onSubmit = ( event ) => {
        event.preventDefault();//Esta funcion sirve para evitar un refresh del navegador web
        if( inputValue.trim().length < 1 ) return;
        // setCategories( cat => [inputValue, ...cat] );
        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

  return (
    <form onSubmit={(event) => onSubmit(event)} aria-label = "form">
        <input 
            type="text" 
            placeholder="Search gifs" 
            value={ inputValue } 
            onChange={ onInputChange }
        ></input>
    </form>
    
  )
}

AddCategory.propTypes = {
    onNewCategory : PropTypes.func.isRequired,
}

