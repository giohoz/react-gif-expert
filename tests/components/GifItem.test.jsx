import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Tests in GifItem', () => {

    const title = "Test";
    const url = "https://sword-art-online.com/sao.jpg";

    test('Should match the snapshot', () => {
        const { container } = render (<GifItem title={title} url={url} ></GifItem>);
        expect(container).toMatchSnapshot();
    });

    test('Should display the image with the URL and ALT indicated.', () => {

        render (<GifItem title={title} url={url} ></GifItem>);
        // expect( screen.getByRole("img").src ).toBe( url );
        // expect( screen.getByRole("img").alt ).toBe( title );
        const { src, alt } = screen.getByRole("img");
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
        
    })

    test('Should display the title in the component', () => {

        render (<GifItem title={title} url={url} ></GifItem>);
        expect( screen.getByText( title ) ).toBeTruthy();
      
    })
    
    
    
  
})
