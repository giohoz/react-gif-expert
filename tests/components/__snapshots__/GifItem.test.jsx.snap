// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifItem Should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test"
      src="https://sword-art-online.com/sao.jpg"
    />
    <p>
       
      Test
       
    </p>
  </div>
</div>
`;
